test/spartan/spec_test.clj

```diff
@@ -139,3 +149,17 @@
                (try (s/assert int? "foo")
                     (catch clojure.lang.ExceptionInfo e
                       (ex-message e))))))
+
+(deftest explain-str-test
+  (s/def ::str string?)
+  (is (= "1 - failed: string? spec: :spartan.spec-test/str\n" (s/explain-str ::str 1))))
+
+(deftest int-in-test
+  (s/def ::int-in (s/int-in 0 10))
+  ;; (is (s/valid? ::int-in 9)) ;; <- This does not pass. It's something to do with (pred x) doesn't work for {:type ::spec}
+  (is (= "11 - failed: (and int? (fn* [%1] (int-in-range? 0 10 %1))) spec: :spartan.spec-test/int-in\n" (s/explain-str ::int-in 11))))
+
+(deftest double-in-test
+  (s/def ::double-in (s/double-in :min 2.5 :max 5.1))
+  ;; (is (s/valid? ::double-in 2.6)) ;; <- This does not pass. It's something to do with (pred x) doesn't work for {:type ::spec}
```

Comment:
I am happy to file a new bug report for this if that's helpful.

Reply:
New issues welcome!